script.js

```diff
@@ -0,0 +1,79 @@
+const $ = (selector) => document.querySelector(selector);
+const $$ = (selector) => document.querySelectorAll(selector)
+
+let currentCountries = {}
+
+const getAllCountries = async function () {
+  const response = await fetch('CountriesData.json')
+  return await response.json()
+}
+
+const displayCountries =  function (data) {
+  const countriesGrid = $('.countries-grid');
+  countriesGrid.innerHTML = ''
+
+  data.forEach(country => {
+    countriesGrid.innerHTML +=
+      `<a onClick="openDetailsPage('${country.name}')"  class="country scale-effect" data-country-name='${country.name}'>
+      <div class="country-flag">
+      <img src="${country.flag}" />
+      </div>
+
+      <div class="country-info">
+      <h2 class="country-title">${country.name}</h2>
+      <ul class="country-brief">
+        <li>
+        <strong>Population: </strong>
+        ${country.population.toLocaleString()}
+        </li>
+        <li>
+        <strong>Region: </strong>
+        ${country.region}
+        </li>
+        <li>
+        <strong>Capital: </strong>
+        ${country.capital}
+        </li>
+      </ul>
+      </div>
+    </a>`
+  });
+}
+
+const openDropDown = async function () {
+  $('.dropdown-body').classList.toggle("on")
+
+  $$('.li-drop').forEach(countryLi => {
+    countryLi.addEventListener("click", async () => {
+      $('.dropdown-header span').textContent = countryLi.textContent
```

Comment:
Are you sure that you want to add event listeners to all of the ul items for every menu click? Why not add them just once at the beginning?

Also, I don't see that you're cleaning up those listeners. It's very important to remove any unnecessary event listeners later.